src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/UseStartsWithInsteadOfIndexOfComparisonWithZero.cs

```diff
@@ -0,0 +1,91 @@
+            category: DiagnosticCategory.Performance,
+            ruleLevel: RuleLevel.IdeSuggestion,
+            description: CreateLocalizableResourceString(nameof(UseStartsWithInsteadOfIndexOfComparisonWithZeroDescription)),
+            isPortedFxCopRule: false,
+            isDataflowRule: false
+            );
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            context.RegisterCompilationStartAction(context =>
+            {
+                var stringType = context.Compilation.GetSpecialType(SpecialType.System_String);
+                if (stringType.GetMembers("StartsWith").FirstOrDefault() is not IMethodSymbol ||
+                    stringType.GetMembers("IndexOf").FirstOrDefault() is not IMethodSymbol indexOfSymbol)
+                {
+                    return;
+                }
+
+                context.RegisterOperationAction(context =>
+                {
+                    var binaryOperation = (IBinaryOperation)context.Operation;
+                    if (binaryOperation.OperatorKind is not (BinaryOperatorKind.Equals or BinaryOperatorKind.NotEquals))
+                    {
+                        return;
+                    }
+
+                    if (IsIndexOfComparedWithZero(binaryOperation.LeftOperand, binaryOperation.RightOperand, indexOfSymbol, out var instanceLocation, out var argumentLocation) ||
+                        IsIndexOfComparedWithZero(binaryOperation.RightOperand, binaryOperation.LeftOperand, indexOfSymbol, out instanceLocation, out argumentLocation))
+                    {
+                        var properties = ImmutableDictionary<string, string?>.Empty;
+                        if (binaryOperation.OperatorKind == BinaryOperatorKind.NotEquals)
+                        {
+                            properties = properties.Add(ShouldNegateKey, "");
+                        }
+
+                        context.ReportDiagnostic(binaryOperation.CreateDiagnostic(Rule, additionalLocations: ImmutableArray.Create(instanceLocation, argumentLocation), properties: properties));
+                    }
+
+                }, OperationKind.Binary);
+            });
+        }
+
+        private static bool IsIndexOfComparedWithZero(IOperation left, IOperation right, IMethodSymbol indexOfSymbol, [NotNullWhen(true)] out Location? instanceLocation, [NotNullWhen(true)] out Location? argumentLocation)
+        {
+            if (right.ConstantValue is not { HasValue: true, Value: 0 } ||
+                left is not IInvocationOperation invocation ||
+                invocation.Arguments.Length != 1 ||
```

Comment:
>invocation.Arguments.Length != 1  [](http://example.com/codeflow?start=16&length=33)

Can we include the case `IndexOf(String, StringComparison)` which taking 2 arguments too?

Reply:
Sure. Will fix this.